Date: 2005-06-22
Initial Package Version: 2.9.6
Upstream Status: http://bugs.archlinux.org/index.php?do=details&id=2891
Origin: Miklos Vajna <vmiklos@frugalware.org>
Description: Adds the rodepends() directive to pacman buildscripts.
diff -Naur pacman-2.9.6.orig/doc/makepkg.8.in pacman-2.9.6/doc/makepkg.8.in
--- pacman-2.9.6.orig/doc/makepkg.8.in	2004-08-04 07:57:17.000000000 +0200
+++ pacman-2.9.6/doc/makepkg.8.in	2005-06-22 21:09:21.000000000 +0200
@@ -303,6 +303,14 @@
 to run).  Packages in this list should follow the same format as \fIdepends\fP.
 
 .TP
+.B rodepends \fI(array)\fP
+An array of packages that this package depends on to run (ie, not required to
+build). Generally \fIrodepends\fP should be avoided in favour of \fIdepends\fP
+except where this will create circular dependency chains. (For example building
+logrotate doesn't requires to have dcron installed.) Packages in this list
+should follow the same format as \fIdepends\fP.
+
+.TP
 .B conflicts \fI(array)\fP 
 An array of packages that will conflict with this package (ie, they cannot both
 be installed at the same time).  This directive follows the same format as
diff -Naur pacman-2.9.6.orig/scripts/makepkg pacman-2.9.6/scripts/makepkg
--- pacman-2.9.6.orig/scripts/makepkg	2005-06-19 15:14:59.000000000 +0200
+++ pacman-2.9.6/scripts/makepkg	2005-06-22 20:52:01.000000000 +0200
@@ -314,7 +314,7 @@
 fi
 
 unset pkgname pkgver pkgrel pkgdesc url license groups provides md5sums force
-unset replaces depends conflicts backup source install makedepends
+unset replaces depends conflicts backup source install makedepends rodepends
 umask 0022
 unset LC_ALL LANG
 
@@ -763,6 +763,9 @@
 for it in "${depends[@]}"; do
 	echo "depend = $it" >>.PKGINFO
 done
+for it in "${rodepends[@]}"; do
+	echo "depend = $it" >>.PKGINFO
+done
 for it in "${removes[@]}"; do
 	echo "remove = $it" >>.PKGINFO
 done
diff -Naur pacman-2.9.7.orig/scripts/gensync pacman-2.9.7/scripts/gensync
--- pacman-2.9.7.orig/scripts/gensync	2005-10-30 01:07:33.000000000 +0200
+++ pacman-2.9.7/scripts/gensync	2005-10-30 01:05:22.000000000 +0200
@@ -143,9 +143,9 @@
 	fi
 	# depends
 	: >depends
-	if [ ${#depends[*]} -gt 0 ]; then
+	if [ ${#depends[*]} -gt 0 -o ${#rodepends[*]} -gt 0 ]; then
 		echo "%DEPENDS%" >>depends
-		for it in "${depends[@]}"; do
+		for it in "${depends[@]}" "${rodepends[@]}"; do
 		  echo "$it" >>depends
 		done
 		echo "" >>depends
diff -Naur pacman-2.9.7.orig/scripts/updatesync pacman-2.9.7/scripts/updatesync
--- pacman-2.9.7.orig/scripts/updatesync	2005-10-30 01:07:33.000000000 +0200
+++ pacman-2.9.7/scripts/updatesync	2005-10-30 01:05:26.000000000 +0200
@@ -146,9 +146,9 @@
 	fi
 	# depends
 	: >depends
-	if [ ${#depends[*]} -gt 0 ]; then
+	if [ ${#depends[*]} -gt 0 -o ${#rodepends[*]} -gt 0 ]; then
 		echo "%DEPENDS%" >>depends
-		for it in "${depends[@]}"; do
+		for it in "${depends[@]}" "${rodepends[@]}"; do
 		  echo "$it" >>depends
 		done
 		echo "" >>depends
